### PR TITLE
Modify media function, fixes type issues

### PR DESCRIPTION
**Pull request by rcantin-w**
## Who is this for?
Devs

## What is it doing for them?
When we enforce types, the media function declared in the theme errors. We decided to change how it was used and align queries across (sometimes it is manually written, sometimes it uses that media function from the theme)

Related to #8566 

## Notes
So if the styling is basic, without needing to access props, you can write it like this
```
  ${props => props.theme.media('medium')`
    grid-template-columns: 3fr 2fr;
  `}
```

or add parentheses
```
  ${props => props.theme.media('medium')(`
    grid-template-columns: 3fr 2fr;
  `)}
```

But if do need access to props, then the parentheses are required in order to work.
```
    ${props =>
      props.theme.media('medium')(`
      background: linear-gradient(
        ${props.theme.color(props.pageBackground)} 200px,
        ${props.theme.color('neutral.700')} 200px
      );
    `)}
```

I went without for simple ones in my alignment but I'd understand wanting to keep it the same everywhere too.

Also ensure you don't use `props =>` inside the function like so, like anywhere else, it'll break. I specify it because the old technique did it like that. You shouldn't find leftovers, but fyi!

```
    ${props =>
      props.theme.media('medium')(`
      background: linear-gradient(
        ${props => props.theme.color(props.pageBackground)} 200px, // No
        ${props.theme.color('neutral.700')} 200px // Yes
      );
    `)}
 ```
 
 There are still a few manual media queries that didn't fit our sizes, I'm not touching them as they might be needed and this isn't about stopping edge cases, but moving forward, using the same method!

### CODE REVIEW

File: content/webapp/components/ImageGallery/ImageGallery.tsx [hunk 1]
```diff
@@ -139,33 +136,18 @@ const Gallery = styled.div.attrs({
...
-      `}
+      ${props.isStandalone && `top: 0;`}
 
-      ${props =>
-        props.isActive &&
-        `
-        opacity: 0.1;
-      `}
-    }
+      ${props.isActive && `opacity: 0.1;`}
```
Comment (alexwlchan) [inline content/webapp/components/ImageGallery/ImageGallery.tsx:145]: Do we need these properties for the medium breakpoint? It looks like they're replicating properties that are applied at any breakpoint on L148 and L150?

File: common/views/themes/default.ts [hunk 2]
```diff
@@ -62,23 +57,15 @@ function makeSpacePropertyValues(
...
+const media =
+  (sizeLabel: Size, minOrMaxWidth: 'min-width' | 'max-width' = 'min-width') =>
+  (styles: TemplateStringsArray | string): string => {
+    if (minOrMaxWidth === 'max-width') {
+      return `@media (max-width: ${themeValues.sizes[sizeLabel]}px) {${styles}}`;
     }
-  `;
-  return acc;
-}, {} as Record<Size, (...args: MediaMethodArgs) => string>);
+    return `@media (min-width: ${themeValues.sizes[sizeLabel]}px) {${styles}}`;
```
Comment (alexwlchan) [inline common/views/themes/default.ts:67]: Could this be simplified as

```
return `@media (${minOrMaxWidth}: ${themeValues.sizes[sizeLabel]}px) {${styles}}`;
```

**Review by alexwlchan — APPROVED** [review]
🧷 Looks good!

File: common/views/themes/default.ts (hunk 2 above)
Comment (rcantin-w) [reply common/views/themes/default.ts:67]: good idea :)

**Pull request merged**